lib/js_routes.rb

```diff
@@ -216,13 +216,19 @@ def build_js(route, parent_route)
   end
 
   def route_js_arguments(route, parent_spec)
+    parts = route.parts
     required_parts = route.required_parts.clone
-    optional_parts = route.parts - required_parts
-    default_parts = route.defaults.select do |part, _|
+    optional_parts = parts - required_parts
+    default_options = route.defaults.select do |part, _|
       FILTERED_DEFAULT_PARTS.exclude?(part) && URL_OPTIONS.include?(part) || required_parts.include?(part)
     end
     [
-      required_parts, optional_parts, serialize(route.path.spec, parent_spec), default_parts
+      parts,
+      required_parts,
+      optional_parts,
+      default_options,
+      Rails.version < '4.0',
+      serialize(route.path.spec, parent_spec)
```

Comment:
There's also an option to pass a table like `{"opt1":0,"arg1":1,"opt2":0,"arg2":1}` (I used `0` in `1` because they're shorter than boolean literals) but as I know JavaScript objects don't maintain the order of fields which is crucial here.